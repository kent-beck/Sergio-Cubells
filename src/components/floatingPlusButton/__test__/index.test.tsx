import { fireEvent, render, screen } from "@testing-library/react";
import FloatingPlusButton from "..";

test("Render a floating plus button", () => {
  const tree = render(<FloatingPlusButton onClick={() => {}} />).asFragment();
  expect(tree).toMatchSnapshot();
});

describe("onClick Floating button", () => {
  const onClickSpy = jest.fn();

  test("onClick is called", () => {
    render(<FloatingPlusButton onClick={onClickSpy} />);
    const floatingPlusButton = screen.getByTestId("floatingPlusButton");
    fireEvent.click(floatingPlusButton);

    expect(onClickSpy).toBeCalled();
  });

  test("onClick is called 3 times", () => {
    render(<FloatingPlusButton onClick={onClickSpy} />);
    const floatingPlusButton = screen.getByTestId("floatingPlusButton");

    fireEvent.click(floatingPlusButton);
    fireEvent.click(floatingPlusButton);
    fireEvent.click(floatingPlusButton);

    expect(onClickSpy).toBeCalledTimes(3);
  });
});
