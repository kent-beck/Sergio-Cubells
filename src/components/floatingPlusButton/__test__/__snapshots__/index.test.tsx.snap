// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a floating plus button 1`] = `
<DocumentFragment>
  <div
    class="floatingPlusButton"
    data-testid="floatingPlusButton"
  />
</DocumentFragment>
`;
